sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-part-time-integration.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService, SshService } from "../../../services";
+import { ESDCIntegrationConfig } from "../../../types";
+import {
+  getDisbursementAmountByValueCode,
+  getGenderCode,
+  getPTMaritalStatusCode,
+  getTotalDisbursementAmount,
+  round,
+} from "../../../utilities";
+import {
+  RecordTypeCodes,
+  CSGD,
+  CSGP,
+  CSGPT,
+} from "./models/e-cert-part-time-integration.model";
+import { SFTPIntegrationBase } from "../../../services/ssh/sftp-integration-base";
+import { FixedFormatFileLine } from "../../../services/ssh/sftp-integration-base.models";
+import { ECertPartTimeFileHeader } from "./e-cert-files/e-cert-file-header";
+import { ECertPartTimeFileFooter } from "./e-cert-files/e-cert-file-footer";
+import { ECertPartTimeFileRecord } from "./e-cert-files/e-cert-file-record";
+import { DisbursementValueType } from "../../../database/entities";
+import { Award, ECertRecord } from "../e-cert-integration-model";
+
+/**
+ * Manages the file content generation and methods to
+ * upload/download the files to SFTP.
+ */
+@Injectable()
+export class ECertPartTimeIntegrationService extends SFTPIntegrationBase<void> {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+
+  constructor(config: ConfigService, sshService: SshService) {
+    super(config.getConfig().zoneBSFTP, sshService);
+    this.esdcConfig = config.getConfig().ESDCIntegration;
+  }
+
+  /**
+   * Create the ECert file content, by populating the
+   * header, detail and trailer records.
+   * @param ecertRecords student, User and application data.
+   * @param fileSequence unique file sequence.
+   * @returns complete ECert content to be sent.
+   */
+  createRequestContent(
+    ecertRecords: ECertRecord[],
+    fileSequence: number,
+  ): FixedFormatFileLine[] {
+    const fileLines: FixedFormatFileLine[] = [];
+    // Header record
+    const header = new ECertPartTimeFileHeader();
+    header.recordTypeCode = RecordTypeCodes.ECertHeader;
+    header.processDate = new Date();
+    fileLines.push(header);
+
+    // Detail records
+    // Calculated values
+    const fileRecords = ecertRecords.map((ecertRecord) => {
+      // ! All dollar values must be rounded to the nearest integer (0.5 rounds up)
+      const roundedAwards = ecertRecord.awards.map(
+        (award) =>
+          ({
+            valueType: award.valueType,
+            valueCode: award.valueCode,
+            valueAmount: round(award.valueAmount).toString(),
+          } as Award),
+      );
+
+      const totalCSGP_PTAmount = getDisbursementAmountByValueCode(
+        roundedAwards,
+        CSGD,
```

Comment:
Instead of using the codes like _CSGD_ we should be using the generic values from the enum `DisbursementValueType`.
Is there something preventing us from doing it?

Reply:
These are not ValueType, these are actual ValueCodes and for PartTime we require seperate amounts for codes.

Reply:
Let's keep it as it is 😉